xen/arch/arm/coproc/coproc.c

```diff
@@ -322,7 +362,7 @@ struct coproc_device *coproc_alloc(struct dt_device_node *np,
         return ERR_PTR(-ENOMEM);
     }
     coproc->dev = dev;
-
+    coproc->driver_features = driver_features;
```

Comment:
I think you need to take into the account flags cross impact. What would you do if you have need_iommu + not_need_sched? If such a case need_iommu doesn't make any sense, since switching code is in a scheduler. As a minimum you need to put a warn about flags mismatch.

Reply:
no, I don't think so: this is up to the driver to decide what and how it needs. if it requests wrong combination of flags it means it has a bug. just imagine we have 10 flags here... what will be the logic to check